File: Data Structures & Algorithms/jump-game-ii/submission-3.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int now_len=0;//第一格可以摸到的最遠距離
        int len=0;//最遠距離
        int jump=0;//跳躍次數
        if(nums.size()<=1) return 0;
        for(int i=0;i<nums.size()-1;i++){
            len=max(len,i+nums[i]);
            //一格一格往前探，看最遠摸到哪裡
            if(i==now_len){
                //如果目前位置已經碰到第一格可以碰到的最遠位置，那jump加一
                //並把該次步數可碰到的最遠距離更新成目前最遠距離
                jump++;
                now_len=len;
            }
        }
        return jump;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left=0
        right=len(heights)-1
        max_water=0
        while left<right:
            current_water=(right-left)*min(heights[left],heights[right])
            max_water=max(max_water,current_water)

            if(heights[left]<heights[right]):
                left+=1
            else:
                right-=1
        return max_water